### ISSUE TITLE
Certificate renewal fails.

### ISSUE THREAD

**Issue by AlastairGrowcott** [smallstep/certificates#200]
### Certificate renewal fails.
I set up a step-ca server and tried creating/renewing certificates using certbot. When that failed I followed the guide to use step-cli on the client. Creating a certificate is fine, but a test renewal fails.

### Your environment
* Ubuntu 18 in a docker container.

### Steps to reproduce
Dockerfile says:

```
RUN cd /root && wget https://github.com/smallstep/cli/releases/download/v0.14.0-rc.3/step-cli_0.14.0-rc.3_amd64.deb
RUN cd /root && dpkg -i step-cli_0.14.0-rc.3_amd64.deb
RUN export HOME=/root && step ca bootstrap --ca-url https://ca.dynautics.local --fingerprint "d931a40160333090ccd090335928168970ff8d706e2cb0c850ae0681eb14721f"
```

When the container starts it runs a script which does the following (I actually ran this manually from within the docker container).

```
cd /root && step ca certificate phabricator.dynautics.local srv.crt srv.key --kty=RSA --standalone --http-listen=:8888   ## Docker forwards port 80 on the host to port 8888. In future it would be nice if the step-ca server could use a configurable port for challenges.
# The step above succeeds:
#        ✔ Provisioner: acme-smallstep (ACME)
#        Using Standalone Mode HTTP challenge to validate phabricator.dynautics.local .. done!
#        Waiting for Order to be 'ready' for finalization .. done!
#        Finalizing Order .. done!
#        ✔ Certificate: srv.crt
#        ✔ Private Key: srv.key

cd /root && step ca renew srv.crt srv.key
```

### Expected behaviour
I'm not really sure exactly what should happen, but basically the certificate should be renewed.

### Actual behaviour
The step-cli reports:

error renewing certificate: client.Renew; client POST https://ca.dynautics.local/renew failed: Post https://ca.dynautics.local/renew: remote error: tls: bad certificate

The step server reports:

2020/03/03 11:34:10 /home/travis/.gimme/versions/go1.13.8.linux.amd64/src/net/http/server.go:3054: http: TLS handshake error from 192.168.0.35:46568: tls: failed to verify client's certificate: x509: certificate specifies an incompatible key usage

### Additional context
I have a number of VMs I need to set upworking automated CA on. If you need me to do any additional debugging/testing/etc., please just let me know.

**Comment by dopey** [comment]
I've not seen that error before. Hop in the [Gitter](https://gitter.im/smallstep/community) and let's try and debug from there. First thing would be figuring out what the actual key usage extensions on that cert are. Could probably see that with a `step certificate inspect <url>`

**Comment by AlastairGrowcott** [comment]
This is weird. I just went through 5 of my VMs, including the one that had the problem originally, and the CA renew worked fine on them all.

I don't think I have changed anything since I first set them up, but now they are all working fine. I haven't done any updates or anything. This is really weird.

Anyway, I cannot reproduce the problem and everything is fine so I guess I will close this bug report.
